pkg/i2gw/providers/common/converter.go

```diff
@@ -31,9 +32,21 @@ import (
 
 // ToGateway converts the received ingresses to i2gw.GatewayResources,
 // without taking into consideration any provider specific logic.
-func ToGateway(ingresses []networkingv1.Ingress, options i2gw.ProviderImplementationSpecificOptions) (i2gw.GatewayResources, field.ErrorList) {
+//
+// If a provider wishes to recieve notifications from the common package,
+// it can pass a notifications.NotificationCallback function which can be used
+// to send notifications on behalf of the provider while keeping the logic of
+// ToGateway seperated from the provider.
+func ToGateway(ingresses []networkingv1.Ingress, options i2gw.ProviderImplementationSpecificOptions, notifyOpts ...notifications.NotificationCallback) (i2gw.GatewayResources, field.ErrorList) {
 	aggregator := ingressAggregator{ruleGroups: map[ruleGroupKey]*ingressRuleGroup{}}
 
+	var notify notifications.NotificationCallback
+	if len(notifyOpts) > 0 {
+		notify = notifyOpts[0]
+	} else {
+		notify = noNotifications
+	}
```

Comment:
```suggestion
	switch len(notifyOpts) {
	case 0:
		notify = noNotifications
	case 1:
		notify = notifyOpts[0]
	default:
		return fmt.Errorf("unsupported amount of callbacks provided. Only 0 or 1 callbacks are currently supported")
	}
```
I think we should clearly state that only 0 - 1 callbacks are allowed here. This error can be triggered only by a wrong provider implementation, so we should avoid it to happen. Feel free to rephrase the error message as you prefer and/or re-structure the switch.